### PR TITLE
feat: airflow modern syntax migration

### PR DESCRIPTION
**Pull request by srijan-paul**
A migration that moves airflow legacy syntax to the newer decorator syntax.

/claim #31

### CODE REVIEW

File: .grit/patterns/airflow_decorator_syntax.md [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  `$parent_func` where {
+    // find all nested function defs that are referenced in a task instantiation.
+    // then: 1) add @task decorator, and 2) replace all task instances with the name of the function.
+    $parent_func <: contains bubble($parent_func) function_definition($name) as $task_func where {
+      $parent_func <: contains bubble($name, $task_func) call($arguments) as $call where {
+        $arguments <: contains keyword_argument(name=`python_callable`, value=$name),
+        $call <: collect_kwargs($kwargs),
+        $decorator_args = join(list=$kwargs, separator=", "),
+        $task_func => `@task($decorator_args)\n$task_func`,
+        // $call => $name
```
Comment (morgante) [inline .grit/patterns/airflow_decorator_syntax.md]: No commented out code please.

Comment (srijan-paul) [reply .grit/patterns/airflow_decorator_syntax.md]: fixed.

File: .grit/patterns/airflow_decorator_syntax.md [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+  }
+}
+
+// collect all references to the variable 
+pattern collect_dag_var_refs($dag_refs) {
+  `$function` where {
+    $function <: dag_definition($var) as $decl,
+    $dag_refs = [],
+    $function <: contains bubble ($function, $dag_refs) identifier() as $dag_ref where {
+      $dag_ref <: after $decl,
```
Comment (morgante) [inline .grit/patterns/airflow_decorator_syntax.md]: What's this line meant to do? $decl is unbound, so this doesn't really check anything.

Comment (srijan-paul) [reply .grit/patterns/airflow_decorator_syntax.md]: Whoops, I realized this pattern wasn't used.
it's from an older impl that I got rid of.
Removed it.

**Review by morgante — APPROVED** [review]
Looks good to start. I will need to do some more testing to validate it works successfully on real repos.

**Pull request merged**